src/Authoring/WinRT.SourceGenerator/Generator.cs

```diff
@@ -222,9 +224,20 @@ class WinRTSyntaxReciever : ISyntaxReceiver
     {
         public List<MemberDeclarationSyntax> Declarations = new List<MemberDeclarationSyntax>();
 
+        private bool HasSomePublicTypes(SyntaxNode syntaxNode)
+        {
+            bool hasSomePublicTypes = false;

```

Comment:
This probably should include enums too and delegates?  Should we just look for any public child node?

Reply:
Updated, seems to work fine.